scripts/aggregate-themes/aggregate-themes.js

```diff
@@ -0,0 +1,150 @@
+const fs = require('fs');
+const glob = require('glob');
+const path = require('path');
+const Logger = require('../utils/logger');
+
+const CONFIG = 'theme.config.js';
+const NODE_MODULES = 'node_modules/';
+const OUTPUT = 'aggregated-themes.js';
+const OUTPUT_PATH = path.resolve(process.cwd());
+const DISCLAIMER = fs.readFileSync(path.resolve(__dirname, 'disclaimer.txt'), 'utf8');
+
+/**
+ * Aggregates theme assets into a single file.
+ *
+ * By default the theme will recursively search all dependencies.
+ * If a theme directory contains a root-theme.scss only that single file will be aggregated.
+ */
+class ThemeAggregator {
+  /**
+   * Aggregates theme assets.
+   * @returns {string|null} - The output path of the aggregated theme file. Null if not generated.
+   */
+  static aggregate() {
+    const defaultConfig = path.resolve(process.cwd(), CONFIG);
+
+    if (fs.existsSync(defaultConfig)) {
+      // eslint-disable-next-line global-require, import/no-dynamic-require
+      return ThemeAggregator.aggregateTheme(require(defaultConfig));
+    }
+
+    return null;
+  }
+
+  /**
+   * Aggregates theme assets.
+   * @param {Object} options - The aggregation options.
+   * @returns {string} - The output path of the aggregated theme file.
+   */
+  static aggregateTheme(options = {}) {
+    ThemeAggregator.validate(options);
+
+    const { theme } = options;
+
+    const dirs = ThemeAggregator.find(`**/themes/${theme}/`, options);
+    const assets = ThemeAggregator.filter(dirs, options);
+
+    if (fs.existsSync(`${NODE_MODULES}${theme}/`)) {
+      assets.unshift(theme);
+    }
+
+    return ThemeAggregator.writeFile(assets, options);
+  }
+
+  /**
+   * Filters theme assets.
+   * @param {string[]} patterns - An array of directory and file paths.
+   * @param {Object} options - The aggregation options.
+   * @returns {string[]} - An array of file names.
+   */
+  static filter(patterns, options) {
+    const assets = [];
+    patterns.forEach((asset) => {
```

Comment:
Just out of curiousity. It feels like we could just do a patterns.map here?

Reply:
The directory filter returns an array that we need to flatten. Map would potentially result in an array of arrays.

Reply:
Thanks! I was looking at this, and felt like I was missing something I couldn't think of.